File: ios/Plugin/Handlers/ProviderHandler.swift
import Foundation
import Capacitor
import FirebaseAuth

protocol ProviderHandler {
    func initialize(plugin: CapacitorFirebaseAuth) -> Void
    func signIn(call: CAPPluginCall) -> Void
    func signOut() throws -> Void
    func isAuthenticated() -> Bool
    func fillResult(credential: AuthCredential?, data: PluginResultData) -> PluginResultData
}

File: android/src/main/java/com/baumblatt/capacitor/firebase/auth/handlers/ProviderHandler.java
package com.baumblatt.capacitor.firebase.auth.handlers;

import android.content.Intent;

import com.baumblatt.capacitor.firebase.auth.CapacitorFirebaseAuth;
import com.getcapacitor.JSObject;
import com.getcapacitor.PluginCall;
import com.google.firebase.auth.AuthCredential;
import com.google.firebase.auth.FirebaseUser;

public interface ProviderHandler {
    void init(CapacitorFirebaseAuth plugin);

    void signIn(PluginCall call);

    void signOut();

    int getRequestCode();

    void handleOnActivityResult(int requestCode, int resultCode, Intent data);

    boolean isAuthenticated();

    void fillResult(AuthCredential credential, JSObject jsResult);
}
